internal/provider/service_resource.go

```diff
@@ -805,16 +824,18 @@ func (r *ServiceResource) updateAllowListState(plan *ServiceResourceModel, state
 }
 
 func (r *ServiceResource) updateServiceStorage(ctx context.Context, plan *ServiceResourceModel, state *ServiceResourceModel, resp *resource.UpdateResponse) {
-	if plan.Storage.ValueInt64() != state.Storage.ValueInt64() || plan.VolumeIOPS.ValueInt64() != state.VolumeIOPS.ValueInt64() {
+	if plan.Storage.ValueInt64() != state.Storage.ValueInt64() || plan.VolumeIOPS.ValueInt64() != state.VolumeIOPS.ValueInt64() || plan.VolumeThroughput.ValueInt64() != state.VolumeThroughput.ValueInt64() {
 		tflog.Info(ctx, "Updating storage size for the service", map[string]interface{}{
-			"id":        state.ID.ValueString(),
-			"from":      state.Storage.ValueInt64(),
-			"to":        plan.Storage.ValueInt64(),
-			"iops_from": state.VolumeIOPS.ValueInt64(),
-			"iops_to":   plan.VolumeIOPS.ValueInt64(),
+			"id":              state.ID.ValueString(),
+			"from":            state.Storage.ValueInt64(),
+			"to":              plan.Storage.ValueInt64(),
+			"iops_from":       state.VolumeIOPS.ValueInt64(),
+			"iops_to":         plan.VolumeIOPS.ValueInt64(),
+			"throughput_from": state.VolumeThroughput.ValueInt64(),
+			"throughput_to":   plan.VolumeThroughput.ValueInt64(),
```

Comment:
NB: I wonder how this behaves in practice.  If the user specifies `null` but the system defaults it to 125, what would happen on the following execution when the user still has `null` set in their terraform code.  Ideally, they wouldn't see any changes.

Reply:
making a change, enforcing volume_type, volume_iops, and volume_throughput to be declared. 👍